mesonbuild/compilers.py

```diff
@@ -1758,8 +1758,8 @@ def get_include_args(self, path, is_system):
         return ['-I' + path]
 
     def get_warn_args(self, level):
-        if level == '2':
-            return ['-wi']
+        if level == '2' or level == '3':
```

Comment:
Do you think you can find out if the same arguments also work with other D compilers and apply this change there too?